core/src/main/java/org/elasticsearch/index/similarity/BM25SimilarityProvider.java

```diff
@@ -42,7 +44,8 @@ public BM25SimilarityProvider(String name, Settings settings) {
         super(name);
         float k1 = settings.getAsFloat("k1", 1.2f);
         float b = settings.getAsFloat("b", 0.75f);
-        boolean discountOverlaps = settings.getAsBoolean("discount_overlaps", true);
+        Version indexVersion = settings.getAsVersion(IndexMetaData.SETTING_VERSION_CREATED, Version.CURRENT);
```

Comment:
Done. Btw, there is already a method that does exactly what we want: `Version#indexCreated()` which I am using for all call-sites.